### ISSUE TITLE
Relative path Issue when injecting custom jobs into nautobot 

### ISSUE THREAD

**Issue by sslhansen** [nautobot/nautobot#5407]
Its pretty hard to follow the issue template for this bug. Will try my best :) 
Here is the [slack](https://networktocode.slack.com/archives/C01NWPK6WHL/p1709641144928579) thread with discussion providing more info on this issue
### Environment
* Nautobot version (Docker tag too if applicable): Tested on 2.1.0
* Python version: 3.10
* Database platform, version: Postgres, 13
* Middleware(s):

### Steps to Reproduce
Job is using relative path imports.
1. Sync job to our production nautobot instance using the git repo extensibility feature. Job works fine
2. Clone the same repo to local machine
3. Copy the contents of /jobs to JOBS_ROOT in developer environment instead of using git repo extensibility. This breaks

### Expected Behavior
Nautobot should recognize and be able to run the jobs.

### Observed Behavior
When containers are starting up, nautobot fails with:
![image](https://github.com/nautobot/nautobot/assets/56868370/4df58980-4844-4027-add7-fd7f6e1105d3)

Our tree structure of /jobs is:
![image](https://github.com/nautobot/nautobot/assets/56868370/0ce7516d-8322-49a7-825a-eb76583e943a)

**Comment by gsnider2195** [comment]
If you're running this in `JOBS_ROOT` you can add a `__init__.py` to your sub-directories and switch to absolute module names. So instead of `from ..models.racks.delete_full_rack import delete_full_rack` you would use `from models.racks.delete_full_rack import delete_full_rack`

**Comment by gsnider2195** [comment]
Here's the relevant docs https://docs.nautobot.com/projects/core/en/stable/development/jobs/migration/from-v1/?h=jobs_root#jobs-in-jobs_root

**Comment by sslhansen** [comment]
We did this initially, but that created an error when we tried the git import.
![image](https://github.com/nautobot/nautobot/assets/56868370/67ce4987-49d0-4c78-ad93-ebd2a0bad603) 
No module named 'models'. Then @glennmatthews suggested for us to use relative paths.(I am sorry for the ping, thought it might be relevant to include you here).

Unfortunately i can't reproduce the error that happens with absolute paths right now. The reason for that is also quite "fun", i will add that here as well:
First, i injected the jobs via jobs_root. This works.
Then, i injected the jobs via git. This also **SEEMS** to work, but it is actually not hitting the code that originates from git. Its hitting the job that is injected via jobs_root folder, because it can resolve the _models_ package from the jobs_root package(you know better than me how that works).
I know that it's not the git code, because if i push some code to the repo, like a print statement that it should hit, and run the job that originates from the <git_repo.slug> , it does not actually reach the statement

**Comment by sslhansen** [comment]
(i could reproduce it by wiping my data, importing the job via git first and then running it)

**Comment by gsnider2195** [comment]
I think the problem is python doesn't let you do relative imports from other packages and in `JOBS_ROOT` every file and directory is its own package. Can you try doing something like:

- `$JOBS_ROOT`
  - `my_job_git_repo/`
    - `__init__.py`
    - `jobs/` 
      - `__init__.py`
      - `constants.py`
      - `facades/`
        - `__init__.py`
        - `cable_facade.py`
        - `job_hook_facade.py`
        - `rack_facade.py`
      - `models/`
        - `__init__.py`
        - `cables/`
          - `__init__.py`
          - `label_existing_cables.py`
        - `job_hooks/`
          - `__init__.py`
          - `cable_hook.py`
        - `racks/`
          - `__init__.py`
          - `create_new_rack.py`
          - `delete_full_rack.py`
          - `print_rack.py`
      - `utils/`
        - `__init__.py`
        - `dcim_utils.py`

**Comment by sslhansen** [comment]
Thanks! We made it work using relative imports and the structure you suggested here. Both in git and "folder injection".
If anyone else reads this and tries it out, remember to import the folder containing your jobs in `__init__.py`, in our case:  `from . import jobs` in  in the root of the git repository.

I am not sure if i should close this issue. It seems like quite a workaround, so i am not sure if you are looking to fix this by making the git vs folder injection process similar, or if you need to document anything etc. My stuff works, so i am happy for now :)

**Comment by glennmatthews** [comment]
I don't think #5537 will directly fix this issue. Fundamentally we're still treating `JOBS_ROOT` as a folder containing a bunch of distinct Python packages, *not* a Python package itself containing a set of submodules, whereas each distinct Git repository is treated as a self-contained Python package containing a `jobs` submodule.

**Comment by pszulczewski** [comment]
I had a similar issue with relative imports for jobs being installed in `$JOBS_ROOT` as Glenn pointed above `$JOBS_ROOT` is a folder containing a bunch of distinct Python packages, not a Python package itself.

I created `$JOBS_ROOT/__init__py` and copied all jobs to `$JOBS_ROOT/jobs` so they became a package itself and relative imports now work. ;-)
Remember about `__init__.py` under `$JOBS_ROOT` and `$JOBS_ROOT/jobs` in case of default they will be `/opt/nautobot/jobs` and `/opt/nautobot/jobs/jobs`
